proxy/src/main/java/com/velocitypowered/proxy/bossbar/VelocityBossBarManager.java

```diff
@@ -0,0 +1,80 @@
+package com.velocitypowered.proxy.bossbar;
+
+import com.velocitypowered.api.bossbar.BossBar;
+import com.velocitypowered.api.bossbar.BossBarColor;
+import com.velocitypowered.api.bossbar.BossBarManager;
+import com.velocitypowered.api.bossbar.BossBarOverlay;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import net.kyori.text.Component;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class VelocityBossBarManager implements BossBarManager {
+
+  private final Map<UUID, BossBar> bossBars = new ConcurrentHashMap<>();
+
+  @Override
+  public @NonNull BossBar create(
+      @NonNull Component title, @NonNull BossBarColor color, @NonNull BossBarOverlay overlay) {
+    return create(title, color, overlay, 1);
+  }
+
+  @Override
+  public @NonNull BossBar create(
+      @NonNull Component title, @NonNull BossBarColor color, @NonNull BossBarOverlay overlay,
+      float progress) {
+    VelocityBossBar bossBar =
+        new VelocityBossBar(this, title, color, overlay, progress, UUID.randomUUID());
+    bossBars.put(bossBar.getUUID(), bossBar);
+    return bossBar;
+  }
+
+  @Override
+  public @Nullable BossBar create(
+      @NonNull Component title,
+      @NonNull BossBarColor color,
+      @NonNull BossBarOverlay overlay,
+      float progress,
+      @NonNull UUID uuid) {
+    if (bossBars.containsKey(uuid)) {
+      return null;
+    }
+    VelocityBossBar bossBar = new VelocityBossBar(this, title, color, overlay, progress, uuid);
+    bossBars.put(uuid, bossBar);
+    return bossBar;
+  }
+
+  @Override
+  public @Nullable BossBar get(@NonNull UUID uuid) {
+    return bossBars.get(uuid);
+  }
+
+  @Override
+  public boolean remove(@NonNull UUID uuid) {
+    BossBar bar = get(uuid);
```

Comment:
As currently implemented, this is not thread-safe and doesn't conform with how you're supposed to use `ConcurrentHashMap`.

Don't try to be "nice" and clean up after everyone else. Our expectation is that when you're done you need to clean up after yourself. (Besides, what if I want to reuse the same boss bar again later on?)